book-2-mvc/chapters/USER_RELATED_DATA.md

```diff
@@ -0,0 +1,295 @@
+## User Related Data
+
+As part of the last exercise, you made views for creating, editing, and viewing dogs. Currently anyone can create a dog, assign it to any Owner, and view a list of all dogs. For the sake of privacy though, it would be nice if owners could only view or edit their own dogs. It would also be nice if in the dog Create form, we didn't make users enter an owner ID into an input field; instead the server would look for the current logged in owner, and default the dog's OwnerId property to that.
+
+To be able to do this, we need to create a system for authentication and autorization. 
+
+### Authentication vs Authorization
+
+Authentication is the process of determining _who_ a user is. One way this could be determined is when a user logs in using an email/password combination. With that combination, the server can assume who the user is. 
+
+Authorization is the process of determining _what a user has access to_. In our case, we're saying that owners should only be able to view and edit their own dogs. If Bob tries to navigate to `/dogs/edit/3` but the dog with the ID 3 belongs to Patty, Bob should not be authorized to view that page.
+
+**IMPORTANT NOTE**
+
+Creating a secure system for authentication and authorization is both incredibly important and very complex. We'll talk a bit more about safely storing sensitive information later in the course, but for now we're going to skip the password step and instead we're going to implement a lite version of auth where users only enter their email addresses to log in.
+
+## Create a Login Page for Owners
+
+We want owners to be able to authenticate, so let's create them a Login page. We'd like the url for this login form to be `/owners/login` which means that we'll need to eventually create two methods in our `OwnerController` called `Login` (again, we need two because the first is to GET the form and the second is to POST the form).
+
+First create a new file in your models folder and name it `Login.cs`. This is a simple class that will only capture a user's email address when logging in. Add the following code
+
+```csharp
+namespace DogWalker.Models
+{
+    public class Login
+    {
+        public string Email { get; set; }
+    }
+}
+```
+
+Now go into the `OwnerController` and add the following methods
+
+```csharp
+public ActionResult Login()
+{
+    return View();
+}
+
+[HttpPost]
+public async Task<ActionResult> Login(LoginViewModel viewModel)
+{
+
+    Owner owner = GetOwnerByEmail(viewModel.Email);
+
+    if (owner == null)
+    {
+        return Unauthorized();
```

Comment:
Should we return `View()` instead? I suppose that brings up the question of `ModelState` too...